KAA-934 CMocka framework integration

forGGe opened the pull request.

forGGe committed:
KAA-934 Add Cmocka framework into build environment

CMocka is an elegant unit testing framework for C with support for mock
objects. It only requires the standard C library, works on a range of
computing platforms (including embedded) and with different compilers.

Web: https://cmocka.org/

forGGe commented:
Do not review! Merge conflicts must be addressed

forGGe committed:
KAA-934 Cmocka framework integration

CMocka framework integrated with Kaa and appropriate bindings
are created. This will guard existing tests from failing.

When all tests will be rewritten to use new framework,
support definitions and macro will be removed.

rasendubi commented on client/client-multi/client-c/test/platform-impl/test_ext_log_storage_memory.c:
```diff
@@ -1,4 +1,4 @@
-/*
+/**
```
That's not a doxygen comment

rasendubi commented:
Please, rebase on the current master. Newly added test (`test_get_request_kaasync_over_buff`) should be adapted to cmocka.

forGGe committed:
KAA-934 Use standart __func__ variable

Instead of GCC-provided __FUNCTION__ which is not compatible with C99
standart.

forGGe committed:
KAA-934 Adopted new test to cmocka

forGGe commented:
Rebased, fixed test.
/cc @rasendubi @dyosick

dyosick commented on client/client-multi/client-c/test/test_kaa_common.c:
```diff
@@ -26,8 +26,10 @@
+void test_profile_update(void **state)
 {
+    (void)state;
+    (void) state;
```
Duplicates.

forGGe committed:
Update test_kaa_common.c

According to the code review

forGGe committed:
Update test_ext_log_storage_memory.c

According to the core review

forGGe commented:
Review addressed

rasendubi commented:
LGTM
